crcsim/experiment/prepare.py

```diff
@@ -130,90 +137,102 @@ def create_scenarios() -> List:
         "100Compliance": 1.0,
         "80Compliance": 0.8,
     }
+
+    propagate_diagnostic_noncompliance = {
+        "Diagnostic_Noncompliance": True,
+        "NO_Diagnostic_Noncompliance": False,
+    }
     scenarios = []
 
     for fqhc, sreening_rates in initial_compliance.items():
         for compliance, diagnostic_rate in diagnostic_compliance_rates.items():
-            baseline = (
-                Scenario(
-                    name=f"{fqhc}_{compliance}_baseline", params=get_default_params()
-                )
-                .transform(transform_initial_compliance(sreening_rates[0]))
-                .transform(transform_diagnostic_compliance(diagnostic_rate))
-            )
-            scenarios.append(baseline)
-
-            implementation = (
-                Scenario(
-                    name=f"{fqhc}_{compliance}_implementation",
-                    params=get_default_params(),
-                )
-                .transform(transform_initial_compliance(sreening_rates[1]))
-                .transform(transform_diagnostic_compliance(diagnostic_rate))
-            )
-            scenarios.append(implementation)
-
-            # Sensitivity analysis 2. Lower cost for stage III and stage IV initial phase
-            baseline_low_cost = deepcopy(baseline)
-            baseline_low_cost.transform(
-                transform_treatment_cost(
-                    "3", "initial", low_initial_stage_3_treatment_cost
-                )
-            ).transform(
-                transform_treatment_cost(
-                    "4", "initial", low_initial_stage_4_treatment_cost
-                )
-            )
-            baseline_low_cost.name = (
-                f"{fqhc}_{compliance}_baseline_low_initial_treat_cost"
-            )
-            scenarios.append(baseline_low_cost)
-
-            implementation_low_cost = deepcopy(implementation)
-            implementation_low_cost.transform(
-                transform_treatment_cost(
-                    "3", "initial", low_initial_stage_3_treatment_cost
+            for (
+                diagnostic_noncompliance,
+                diagnostic_noncompliance_bool,
```

Comment:
Naming is also a bit confusing here - perhaps `propagation_name` and `propagation_value`?

(It's usually not a good idea to include a variable's type in its name.)